js/semaphore.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * A Semaphore that allows only the execution of one process at a time.
+ *
+ * @see https://gist.github.com/weimingw/a32d070a760dbc92ebb716b51bd44ac2#file-semaphore-js
+ */
+export const semaphore = (maxRunningRequests = 1) => {

```

Comment:
I see your comment of porting it to a function because we dont have classes in commons JS but I think it would be still ok to create one, no? Either way, I still like this.

Reply:
It ends up being the same functionality and making a class for this one opens a precedent I don't know we want open